### PR TITLE
Compiler fixes + TR1 isolation

### PR DESCRIPTION
**Pull request by uartie**
fixes for gcc 6.x and TR1 issues that will become more apparent with libaymi change https://github.com/01org/libyami/pull/573

### CODE REVIEW

File: tests/vppoutputencode.cpp [hunk 1]
```diff
@@ -63,7 +63,7 @@ bool VppOutputEncode::init(const char* outputFileName, uint32_t /*fourcc*/, int
     m_width = width;
     m_height = height;
     m_output.reset(EncodeOutput::create(outputFileName, m_width, m_height));
-    return m_output;
```
Comment (xuguangxin) [inline tests/vppoutputencode.cpp]: this not needed you can refer to http://en.cppreference.com/w/cpp/memory/shared_ptr/operator_bool

File: tests/yamitranscode.cpp [hunk 2]
```diff
@@ -300,7 +300,7 @@ class TranscodeTest
             return false;
         }
         m_allocator = createAllocator(m_output, m_display, m_cmdParam.m_encParams.ipPeriod);
-        return m_allocator;
```
Comment (xuguangxin) [inline tests/yamitranscode.cpp]: it's not needed too

File: tests/vppoutputencode.cpp (hunk 1 above)
Comment (xuguangxin) [inline tests/vppoutputencode.cpp]: My mistake, Now i know c++ 11 do not allow this

File: tests/yamitranscode.cpp (hunk 2 above)
Comment (xuguangxin) [inline tests/yamitranscode.cpp]: my mistake.

File: tests/vppoutputencode.cpp (hunk 1 above)
Comment (uartie) [inline tests/vppoutputencode.cpp]: Right.  C++11 share_ptr bool operator is defined as `explicit` which is why we need this.

**Pull request merged**